cpp/src/rolling/grouped_rolling.cu

```diff
@@ -273,28 +280,29 @@ std::unique_ptr<column> expand_to_column(Calculator const& calc,
   return window_column;
 }
 
-/// Time-range window computation, with
+/// Range window computation, with
 ///   1. no grouping keys specified
-///   2. timetamps in ASCENDING order.
+///   2. rows in ASCENDING order.
 /// Treat as one single group.
-std::unique_ptr<column> time_range_window_ASC(column_view const& input,
-                                              column_view const& timestamp_column,
-                                              TimeT preceding_window,
-                                              bool preceding_window_is_unbounded,
-                                              TimeT following_window,
-                                              bool following_window_is_unbounded,
-                                              size_type min_periods,
-                                              std::unique_ptr<aggregation> const& aggr,
-                                              rmm::cuda_stream_view stream,
-                                              rmm::mr::device_memory_resource* mr)
+template <typename T>
+std::unique_ptr<column> range_window_ASC(column_view const& input,
+                                         column_view const& orderby_column,
+                                         T preceding_window,
+                                         bool preceding_window_is_unbounded,
+                                         T following_window,
+                                         bool following_window_is_unbounded,
+                                         size_type min_periods,
+                                         std::unique_ptr<aggregation> const& aggr,
+                                         rmm::cuda_stream_view stream,
+                                         rmm::mr::device_memory_resource* mr)
 {
   size_type nulls_begin_idx, nulls_end_idx;
-  std::tie(nulls_begin_idx, nulls_end_idx) = get_null_bounds_for_timestamp_column(timestamp_column);
+  std::tie(nulls_begin_idx, nulls_end_idx) = get_null_bounds_for_orderby_column(orderby_column);
```

Comment:
No need to use `std::tie` now :)
```suggestion
  auto [nulls_begin_idx, nulls_end_idx] = get_null_bounds_for_orderby_column(orderby_column);
```

Reply:
(Repeating inline for context.)

I appreciate the suggestion to use structured bindings, but I'm afraid they don't suffice in this context: I would have to use those as captures for the lambdas that follow, and that is unfortunately disallowed.
```
../src/rolling/grouped_rolling.cu(359): error: structured binding cannot be captured
          detected during:
            instantiation of "std::unique_ptr<cudf::column, std::default_delete<cudf::column>> cudf::<unnamed>::grouped_r
ange_rolling_window_impl<OrderByT>(const cudf::column_view &, const cudf::column_view &, const cudf::order &, const rmm::
device_uvector<cudf::size_type> &, const rmm::device_uvector<cudf::size_type> &, cudf::range_window_bounds, cudf::range_w
indow_bounds, cudf::size_type, const std::unique_ptr<cudf::aggregation, std::default_delete<cudf::aggregation>> &, rmm::c
uda_stream_view, rmm::mr::device_memory_resource *) [with OrderByT=int8_t]"
```